src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -566,6 +573,12 @@ type DiagnosticItem =
             | ErrorCode.ArgumentOfUserDefinedTypeInAttribute    -> "Items of user defined type cannot be used as attribute arguments."
             | ErrorCode.TypeParameterizedArgumentInAttribute    -> "The type of attribute arguments must be known at compile time."
             | ErrorCode.AttributeArgumentTypeMismatch           -> "The type of the given argument does not match the expected type."
+            | ErrorCode.InvalidEntryPointPlacement              -> "Invalid entry point. Entry point attributes may only occur on suitable callables without type-parametrizations."
+            | ErrorCode.QubitTypeInEntryPointSignature          -> "Invalid entry point. Values of type Qubit may not be used as arguments or return values to entry points."
+            | ErrorCode.CallableTypeInEntryPointSignature       -> "Invalid entry point. Values of operation or function type may not be used as arguments or return values to entry points."
+            | ErrorCode.UserDefinedTypeInEntryPointSignature    -> "Invalid entry point. Values of user defined type may not be used as arguments or return values to entry points."
+            | ErrorCode.MultipleEntryPoints                     -> "Invalid entry point. An entry point {0} already exists in {1}."
+            | ErrorCode.InvalidEntryPointSpecialization         -> "Entry points cannot have any other specializations besides the default body."
```

Comment:
Is this the error message raised when a user marks as entry point an operation with adjoint or controlled specializations defined?

Reply:
Yes. I am open to suggestions regarding the formulation... I don't know how to make it best comprehensible.